### PR TITLE
Add changelog deprecation notice

### PR DESCRIPTION
**Pull request by shilangyu**
Created out of: https://github.com/lsd-rs/lsd/pull/900#discussion_r1329966830

### CODE REVIEW

**Comment by shilangyu** [comment]
The pull request template still mentions that a changelog entry should be added. Should I remove it?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # Changelog
+
+NOTE: This file is no longer being updated. For release notes follow [GitHub releases](https://github.com/lsd-rs/lsd/releases).
```
Comment (zwpaper) [inline CHANGELOG.md:3]: by saying GitHub auto-generates release notes, I mean https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

We don't have to update this file and will generate by PRs when releasing.

**Comment by zwpaper** [comment]
can you also help drop this line: https://github.com/lsd-rs/lsd/blob/master/.github/PULL_REQUEST_TEMPLATE#L9

File: CHANGELOG.md (hunk 1 above)
Comment (shilangyu) [reply CHANGELOG.md:3]: Not sure what you would like me to do here. I understand what you are trying to achieve, but not what is the future of this CHANGELOG file. What should the notice say? If there should be a notice at all?

**Pull request merged**